docs/src/grids.md

```diff
@@ -0,0 +1,646 @@
+
+grid = RectilinearGrid(architecture,
+                       topology = (Periodic, Periodic, Bounded),
+                       size = (16, 8, 4),
+                       x = (0, 64),
+                       y = (0, 32),
+                       z = (0, 8))
+
+# output
+16×8×4 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 3×3×3 halo
+├── Periodic x ∈ [0.0, 64.0) regularly spaced with Δx=4.0
+├── Periodic y ∈ [0.0, 32.0) regularly spaced with Δy=4.0
+└── Bounded  z ∈ [0.0, 8.0]  regularly spaced with Δz=2.0
+```
+
+This simple grid
+
+* Uses the CPU. To make a grid on the GPU --- which means that computations on the grid will be conducted using a GPU connected to our CPU, if one is available --- we write `architecture = GPU()`.
+* Has a domain that's "periodic" in ``x, y``, but bounded in ``z``. More on what that means, exactly, in a bit.
+* Has `16` cells in `x`, `8` cells in `y`, and `4` cells in `z`. That means there are ``16 \times 8 \times 4 = 512`` cells in all.
+* Has an `x` dimension that spans from `x=0`, to `x=64`. And `y` spans `y=0` to `y=32`, and `z` spans `z=0` to `z=8`.
+* Has cells that are all the same size, dividing the ``16 \times 8 \times 4`` box into ``4 \times 4 \times 2`` cells. Note that length units are whatever is used to construct the grid, so it's up to the user to make sure that all inputs use consistent units.
+
+Next we illustrate the construction of a grid on the _GPU_ (for this to work, a GPU has
+to be available and `CUDA.jl` must be working --- for more about that, see the
+[`CUDA.jl` documentation](https://cuda.juliagpu.org/stable/)).
+The grid is two-dimensional in ``x, z`` (it's missing the `y`-dimension), and has variably-spaced
+cell interfaces in the `z`-direction,
+
+```@setup grids_gpu
+using Oceananigans
+```
+
+```jldoctest grids_gpu
+architecture = GPU()
+z_faces = [0, 1, 3, 6, 10]
+
+grid = RectilinearGrid(architecture,
+                       topology = (Periodic, Flat, Bounded),
+                       size = (10, 4),
+                       x = (0, 20),
+                       z = z_faces)
+
+# output
+10×1×4 RectilinearGrid{Float64, Periodic, Flat, Bounded} on GPU with 3×0×3 halo
+├── Periodic x ∈ [0.0, 20.0)      regularly spaced with Δx=2.0
+├── Flat y
+└── Bounded  z ∈ [0.0, 10.0]      variably spaced with min(Δz)=1.0, max(Δz)=4.0
+```
+
+The `y`-dimension is marked `Flat` by writing `topology = (Periodic, Flat, Bounded)`.
+This means that nothing varies in `y`; derivatives in `y` are 0.
+This also means that the kwarg that specifies the `y`-domains may be omitted, and that
+the `size` has only two elements rather than 3 as in the first example.
+Notice that in the stretched cell interfaces specified by `z_interfaces`, the number of
+vertical cell interfaces is `Nz + 1 = length(z_interfaces) = 5`, where `Nz = 4` is the number
+of cells in the vertical.
+
+## The nuts and bolts of building a grid
+
+These examples illustrate how the definition of any grid requires
```

Comment:
```suggestion
These examples illustrate how the definition of any grid requires:
```

Reply:
confused whether this is correct

Reply:
ok, perhaps it's a style choice -- I'll leave it up to you to decide!

Reply:
I think a colon is incorrect here!

Reply:
(because the bullet points "finish" the sentence started above)

Reply:
gotta